test-two/src/index.js

```diff
@@ -3,21 +3,36 @@ import { getUsersForOrganisation } from "./services/github";
 
 const baseUrl = "https://api.github.com/";
 const organisationId = "google";
-const gitHubApiAuthToken = ""; // add your GitHub API OAuth key here to increase request limit
+const gitHubApiAuthToken = "";
+export const pages = 22; // There are 2144 users so 22 pages are sufficient
```

Comment:
One way I would do that is to use a while loop and run the getUsersForOrganisation function until it returns nothing and then break out of the loop.

```
let responseHasContent = true;
let page = 1;

while (responseHasContent) {
    const users = await getUsersForOrganisation(
      baseUrl,
      gitHubApiAuthToken,
      organisationId,
      page
    );

  if (users.length > 0) {
    const output = users.reduce((result, user) => {
      return `${result}Username: ${user.login}\n`;
    }, "");
    page++;
  } else {
    responseHasContent = false;
  }
}
```

The problem I found with that is that it would then execute the calls in sequence rather than at the same time so it would take longer.